### ISSUE TITLE
Can't use go install due to replace directives in go.mod

### ISSUE THREAD

**Issue by dmorneau** [jthomperoo/custom-pod-autoscaler#160]
**Describe the bug**
go install does not work with custom-pod-autoscaler because its go.mod includes replace directives

**To Reproduce**

`go install github.com/jthomperoo/custom-pod-autoscaler/v2@v2.4.0`

**Expected behavior**
It would be useful if the package was compatible with go install (for context, I'm trying to install it in an existing container).

**Additional context**
```
go: downloading github.com/jthomperoo/custom-pod-autoscaler/v2 v2.4.0
go install: github.com/jthomperoo/custom-pod-autoscaler/v2@v2.4.0 (in github.com/jthomperoo/custom-pod-autoscaler/v2@v2.4.0):
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

https://github.com/jthomperoo/custom-pod-autoscaler/blob/7c8f66d125c3cffcbfdf76665219f2cb70cdb89f/go.mod#L18

**Comment by jthomperoo** [comment]
Unfortunately this project has a dependency on [Argo Rollouts](https://github.com/argoproj/argo-rollouts) which includes a direct dependency on `k8s.io/kubernetes` which requires these replace directives:
https://github.com/argoproj/argo-rollouts/issues/473

I'm looking at removing this direct dependency from this codebase, but in order to support `go install` I'll need to think of a way around this until Argo Rollouts removes this dependency. I don't think that will be any time soon, so the options I can think of are these:

- Copy the required code into this codebase from Argo Rollouts, eliminating the dependency.
- Implement jthomperoo/custom-pod-autoscaler#159 which if it can be implemented will mean this should not be an issue.

I'll keep this issue updated with any progress.

**Comment by jthomperoo** [comment]
This is now available in version [v2.5.0](https://github.com/jthomperoo/custom-pod-autoscaler/releases/tag/v2.5.0).
